File: TestCoverageReport/app/src/main/java/io/github/brunogabriel/testcoveragereport/MainActivity.java
package io.github.brunogabriel.testcoveragereport;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatTextView;
import android.view.View;

/**
 * Created by brunogabriel on 25/01/17.
 */

public class MainActivity extends AppCompatActivity implements View.OnClickListener{
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.button_hide).setOnClickListener(this);
        findViewById(R.id.button_msg).setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        if (view.getId() == R.id.button_hide) {
            findViewById(R.id.button_hide).setVisibility(View.GONE);
        } else {
            ((AppCompatTextView) findViewById(R.id.text_hello))
                    .setText(getString(R.string.hello_world));
        }
    }

    public boolean isTextHelloWorld() {
        return ((AppCompatTextView) findViewById(R.id.text_hello)).getText()
                .toString().equalsIgnoreCase(getResources().getString(R.string.hello_world));
    }
}

File: TestCoverageKotlin/app/src/main/java/io/github/brunogabriel/testcoveragekotlin/MainActivity.kt
package io.github.brunogabriel.testcoveragekotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.annotation.StringRes
import androidx.appcompat.app.AlertDialog
import io.github.brunogabriel.testcoveragekotlin.validator.LoginValidator
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by brunogabriel on 2019-10-09.
 */
class MainActivity : AppCompatActivity(), MainContract.View {
    override lateinit var presenter: MainContract.Presenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        presenter = MainPresenter(this, LoginValidator())
        setupView()
    }

    override fun showLoginSuccess() {
        showMessage(R.string.login_success)
    }

    override fun showLoginFail() {
        showMessage(R.string.login_fail)
    }

    private fun showMessage(@StringRes messageRes: Int) {
        AlertDialog.Builder(this)
            .setMessage(messageRes)
            .setPositiveButton(android.R.string.ok, null)
            .show()
    }

    private fun setupView() {
        setSupportActionBar(toolbar)
        login_button.setOnClickListener {
            presenter.onLoginClicked(
                username_edit_text.text.toString(),
                password_edit_text.text.toString()
            )
        }
    }
}
